### PR TITLE
fix(tuttid): coalesce duplicate provider installs

### PR DESCRIPTION
**Pull request by jomeswang**
## Summary

- coalesce overlapping install actions for the same agent provider inside the daemon
- let concurrent callers share the first install result instead of running a second npm mutation
- retain active-action token isolation for overlapping install and update actions
- add a regression test matching the duplicate Tutti Agent install seen in the daemon logs

Different providers remain independent at the action layer; the existing file lock continues to serialize conflicting npm mutations across processes.

## Verification

- `go test ./services/tuttid/service/agentstatus -count=1`
- `go test -race ./services/tuttid/service/agentstatus -run TestServiceRunActionCoalescesConcurrentInstallsForSameProvider -count=1`
- `pnpm check:changed`
- push hook: `pnpm check:changed -- --push-ready`

## Checklist

- [x] This does not change agent session/turn/goal lifecycle semantics.
- [x] I kept the change focused on one concern.
- [x] No documentation update is needed; the external action contract is unchanged and this fixes duplicate execution.
- [x] No README/CONTRIBUTING language changed.
- [x] I ran the lowest meaningful local checks for the changed surface.
- [x] The commit is signed off with DCO.

---

<a href="https://app.devin.ai/review/tutti-os/tutti/pull/1779" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: services/tuttid/service/agentstatus/service.go [hunk 1]
```diff
@@ -620,30 +626,43 @@ func (s Service) RunAction(ctx context.Context, input RunActionInput) (RunAction
...
+			Result:    actionResult,
 			StartedAt: startedAt,
 		})
-		return result, err
-	case ActionUpdate:
-		return s.runUpdateAction(ctx, spec, result)
-	default:
-		return RunActionResult{}, ErrInvalidAction
+		return actionResult, actionErr
+	})
```
[the remark this answers is not in the document]
Comment (jomeswang) [reply services/tuttid/service/agentstatus/service.go:596]: Addressed in cfeb651cf: the coalesced install now runs with context.WithoutCancel(baseContext(ctx)), so one HTTP caller disconnecting cannot cancel the shared mutation or the readiness reconciler waiting on it. The install remains bounded by the existing installer/probe timeouts. The concurrency test now cancels the leader before releasing the installer and verifies both callers still receive the shared successful result.
